packages/manager-react-components/src/hooks/datagrid/useIcebergV6.tsx

```diff
@@ -24,46 +24,35 @@ export function useResourcesIcebergV6<T = unknown>({
   queryKey,
   defaultSorting = undefined,
 }: IcebergFetchParamsV6 & IcebergV6Hook) {
-  const [totalCount, setTotalCount] = useState(0);
-  const [flattenData, setFlattenData] = useState<T[]>([]);
   const [queryParams, setQueryParams] = useState<QueryParams>({
     pageIndex: 1,
     sorting: { desc: false, id: '' },
   });
   const [sorting, setSorting] = useState<ColumnSort>(defaultSorting);
 
-  const {
-    data,
-    fetchNextPage,
-    isError,
-    isLoading,
-    error,
-    status,
-  } = useInfiniteQuery({
-    initialPageParam: null,
-    queryKey: [...queryKey, sorting],
-    queryFn: ({ pageParam }) =>
-      fetchIcebergV6<T>({
-        route,
-        pageSize,
-        page: pageParam,
-        sortBy: sorting?.id || null,
-        sortReverse: sorting?.desc,
-      }),
-    staleTime: Infinity,
-    retry: false,
-    getNextPageParam: () => queryParams.pageIndex,
-  });
+  const { data, fetchNextPage, isError, isLoading, error, status } =
+    useInfiniteQuery({
+      initialPageParam: null,
+      queryKey: [...queryKey, queryParams.pageIndex, sorting],
+      queryFn: ({ pageParam }) =>
+        fetchIcebergV6<T>({
+          route,
+          pageSize,
+          page: pageParam,
+          sortBy: sorting?.id || null,
+          sortReverse: sorting?.desc,
+        }),
+      staleTime: Infinity,
+      retry: false,
+      getNextPageParam: () => queryParams.pageIndex,
+    });
 
-  useEffect(() => {
-    const flatten = data?.pages.map((page) => page.data).flat();
-    setFlattenData(flatten);
+  const flattenData = useMemo(() => {
+    return data?.pages.map((page) => page.data).flat() ?? [];
```

Comment:
Why not using the `select` built-in function to compute data?

Reply:
indeed it will avoid rerenders if for example the cursor changes but the data remains the same

Reply:
(while using a refetchInterval)

Reply:
and btw it would be nice to be able to pass some options to the useInfiniteQuery

Reply:
I understand for the select. Is a good idea. 

But for the possibility to pass options in `useInfiniteQuery` :  I think it's another subject and this implicate breaking change. And I don't think it's good to include this debate on a bug fix. 

If I can include fast optimization I do it with pleasure. But I scare this change block the validation.

Reply:
But for information @tristanwagner If I'm not mistaken : The rerender is not avoid because the object is regenerated with another reference. 

So the object content is not compared only the reference. And React see a change same it's exactly same data.

Reply:
Oh good to known ! Thank you for this information 😁 I will do it !